src/bci_build/package.py

```diff
@@ -1838,7 +1838,8 @@ def _create_php_bci(
 {DOCKERFILE_RUN} chmod +x /usr/local/bin/docker-entrypoint.sh; \
     ln -s su /usr/bin/gosu; \
     mkdir /docker-entrypoint-initdb.d; \
-    sed -ri "s|^#?(listen_addresses)\s*=\s*\S+.*|\1 = '*'|" /usr/share/postgresql{ver}/postgresql.conf.sample
+    sed -ri "s|^#?(listen_addresses)\s*=\s*\S+.*|\1 = '*'|" /usr/share/postgresql{ver}/postgresql.conf.sample; \
+    mkdir -p "$PGDATA" && chown -R postgres:postgres "$PGDATA" && chmod 0777 "$PGDATA"
```

Comment:
chmod 0777? Why?

Reply:
So that the entrypoint still works when run as a non-root user.

Reply:
I don't understand this. We're chowning it explicitly to the user just before  and it anyway is a volume that has its own inode for this directory, so why does this make any difference?

There's gotta be a better way. Also, as the docker entry point sets up the permission also (if invoked as root), can we not start it as root?

Reply:
I have removed the `chmod 0777`, it didn't fix anything in particular, you still can't run the database process as an arbitrary user, but you can run it as `postgres` without the `chmod`. So from my pov it's not needed.